JavaCollections2_task1/src/main/java/org/Marta/MyMap.java

```diff
@@ -0,0 +1,115 @@
+package org.Marta;
+
+public class MyMap implements OwnMap {
+    Node root;
```

Comment:
Mapę robi się na zwykłej tablicy :D

Reply:
W zadaniu nie było napisane, że to musi być tablica ;)

Reply:
Poza tym, wiele istniejących map w bibliotekach, jest zaimplementowanych jako drzewo czerwono-czarne. Moje to zwykłe BST, ale to moja pierwsza implementacja xD

Reply:
https://en.wikipedia.org/wiki/Hash_table

NO w sumie nie ma tego w treści zadania - masz rację, ale jednak domyślnie chodzi tutaj o standardową mapę. :P Może faktycznie doprecyzuję treść. 

To jest o tyle ważne, że dzięki temu zadaniu jesteś w stanie w miarę zrozumieć o co chodzi z mapą, jak działa hashcode i equals. A to baaardzo ważny koncept.

Reply:
Zobacz jakie są (Average) złożoności 
![image](https://user-images.githubusercontent.com/17989854/118353236-a89b3380-b565-11eb-99b7-4e1232d2e4d9.png)

bo niestety w mapie tez może być przypadek (jeżeli hashcode jest źle napisany), że insert będzie O(n) :) Ale to właśnie jest coś czego się uczy w tym zadaniu - czemu ten hashcode jest tak ważny.